packages/repository/test/unit/repositories/relation.repository.unit.ts

```diff
@@ -108,6 +112,95 @@ describe('relation repository', () => {
     });
   });
 
+  context('resolveHasManyMetadata', () => {
+    it('retains non-resolver type', () => {
+      class TargetModel extends Entity {}
+      const meta: HasManyDefinition = {
+        type: RelationType.hasMany,
+        target: TargetModel,
+      };
+      const result = resolveHasManyMetadata(meta);
+
+      expect(result).to.eql(meta);
+    });
+
+    it('throws if no target relation metadata is found', () => {
+      class TargetModel extends Entity {
+        static definition = new ModelDefinition({
+          name: 'TargetModel',
+        });
+      }
+      const meta: HasManyDefinition = {
+        type: RelationType.hasMany,
+        target: () => TargetModel,
+      };
+      expect(() => resolveHasManyMetadata(meta)).to.throw(
+        /no belongsTo metadata found/,
+      );
+    });
+
+    it('throws if no belongsTo metadata is found', () => {
```

Comment:
Hmm I am a bit confused here: according to the test case, a hasMany relation should always come with a corresponding belongsTo relation, while shouldn't they be decoupled?

Reply:
This test covers the case when a complementing belongsTo relation _is_ defined.

Reply:
And thanks again for explaining the test above!

A brief summary:

The `keyTo` property must be either provided in the parameter of `@hasMany()` decorator, or be inferred from the target model's belongsTo relation.

I feel it would be good to document this behaviour somewhere so people know the correct way to define the `keyTo` property :)